x-pack/test/functional/apps/ml/permissions/full_ml_access.ts

```diff
@@ -358,10 +361,16 @@ export default function ({ getService }: FtrProviderContext) {
             await ml.dataVisualizerIndexBased.assertDataVisualizerTableExist();
 
             await ml.testExecution.logTestStep(
-              'should display the actions panel with Discover card'
+              `should display the actions panel ${
+                testUser.discoverAvailable ? 'with' : 'without'
+              } Discover card`
             );
             await ml.dataVisualizerIndexBased.assertActionsPanelExists();
-            await ml.dataVisualizerIndexBased.assertViewInDiscoverCardExists();
+            if (testUser.discoverAvailable) {
```

Comment:
Nit - @pheyos this block of code is repeated several times, so we could add something like `assertViewInDiscoverCard(exists: boolean)`  to the `dataVisualizerIndexBased` service. But otherwise these changes LGTM

Reply:
Done in 0791551